src/core/core_tags.js

```diff
@@ -0,0 +1,410 @@
+                        tagsRest = tags.slice(1);
+                        if (tagsRest.length === 1)
+                            tagsRest = tagsRest[0];
+
+                        for(n = 0; n < this._index[tags[0]].list.length; n++) {
+                            item = this._index[tags[0]].list[n];
+                            if ((this._key ? ! index[item[this._key]] : (items.indexOf(item) === -1)) && item.tags.has(tagsRest)) {
+                                if (this._key)
+                                    index[item[this._key]] = true;
+                                items.push(item);
+                            }
+                        }
+
+                        continue;
+                    }
+                }
+
+                if (tag && typeof(tag) === 'string' && this._index[tag]) {
+                    for(n = 0; n < this._index[tag].list.length; n++) {
+                        item = this._index[tag].list[n];
+
+                        if (this._key) {
+                            if (! index[item[this._key]]) {
+                                index[item[this._key]] = true;
+                                items.push(item);
+                            }
+                        } else if (items.indexOf(item) === -1) {
+                            items.push(item);
+                        }
+                    }
+                }
+            }
+
+            return items;
+        }
+    };
+
+
+
+    /**
+    * @name pc.Tags
+    * @class Set of tag names
+    * @constructor Create an instance of an Tags.
+    * @param {Object} parent (optional) Parent object who tags belong to.
+    * Note: Tags are used as addition of `pc.Entity` and `pc.Asset` as `tags` field.
+    */
+
+    /**
+     * @event
+     * @name pc.Tags#add
+     * @param {String} tag Name of a tag added to a set.
+     * @param {Object} parent Parent object who tags belong to.
+    */
+
+    /**
+     * @event
+     * @name pc.Tags#remove
+     * @param {String} tag Name of a tag removed from a set.
+     * @param {Object} parent Parent object who tags belong to.
+    */
+
+    /**
+     * @event
+     * @name pc.Tags#change
+     * @param {Object} parent Parent object who tags belong to.
+     * @description fires when any change add/remove been made.
+     * It will fire once on bult changes, while `add`/`remove` will fire on each tag operation
+    */
+
+    var Tags = function(parent) {
+        this._index = { };
+        this._list = [ ];
+        this._parent = parent;
+
+        pc.events.attach(this);
+    };
+
+    Tags.prototype = {
+        /**
+        * @function
+        * @name pc.Tags#add
+        * @description Add a tag, duplicates are ignored. Can be array or comma separated arguments for multiple tags.
+        * @param {String} name of a tag, or array of tags
+        * @returns {Boolean} true if any tag were added
+        * @example
+        * tags.add('level-1');
+        * @example
```

Comment:
Is this how the JSdoc example tag works? I thought you could only have one. Maybe I'm wrong - I never tried using multiple...